libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -1044,33 +1044,44 @@ class MapboxNavigation @VisibleForTesting internal constructor(
         restartRouteRefreshScope()
         threadController.getMainScopeAndRootJob().scope.launch(Dispatchers.Main.immediate) {
             routeUpdateMutex.withLock {
-                historyRecordingStateHandler.setRoutes(routes)
                 val routesSetResult: Expected<RoutesSetError, RoutesSetSuccess>
-                when (val processedRoutes = setRoutesToTripSession(routes, setRoutesInfo)) {
-                    is NativeSetRouteValue -> {
-                        val directionsSessionRoutes = Utils.createDirectionsSessionRoutes(
-                            routes,
-                            processedRoutes,
-                            setRoutesInfo
+                if (setRoutesInfo is SetRoutes.Alternatives &&
+                    routes.first() != directionsSession.routes.firstOrNull()) {
```

Comment:
What's the performance of the full equals function for very long routes? With this addition we'd be running it twice for each route set operation. Would using `route#id` here be enough?

Reply:
> Would using route#id here be enough?

I think it would. Why do we bother comparing the whole route in the first place then?